### ISSUE TITLE
Empirical evidence that Synapse didn't send device deletions over federation

### ISSUE THREAD

**Issue by turt2live** [matrix-org/synapse#8459]
## Description

*Backstory*: There's an encrypted watercooler room where people have verified each other to ensure the room stays as a green shield at all times.

*I'm fairly sure this is a duplicate, but can't find anything.*

I added 2 encryption-capable devices to my account and then promptly deleted them some time later on Synapse v1.20.0 - my local server figured it out, but everyone in the room reported that the devices were still there (on other homeservers).

After they ran `insert into device_lists_remote_resync (user_id, added_ts) values ('@travis:t2l.io', (extract(epoch from now())::bigint * 1000));` it fixed itself.

### Steps to reproduce

- Have an account and be verified with someone
- Log in to an unverified device
- Let the alarm bells go off
- Log out or delete that device
- Alarm bells are still going off

### Version information

- **Version**: 1.20.0
- **Install method**: pip
- **Platform**: ubuntu box somewhere in the cloud

**Comment by clokep** [comment]
I did some testing with this and was able to reproduce some weird behavior on v1.20.0 (I actually ended up with the opposite issue you had -- the federated server never realized there was a second device). I was unable to reproduce this with v1.21.0 so I'm hoping some of the federation catch-up fixes are coming into play here (#8230, #8247, #8258, #8272, #8322).

It would be great if this could be re-tested on v1.21.0 once that is released!
